spec/realtime/auth.test.js

```diff
@@ -517,6 +517,37 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		authUrl: echoServer + '/respondwith?status=403'
 	}, true); /* expectFailed: */
 
+	exports.authUrl_403_previously_active = function(test) {
+		var realtime, rest = helper.AblyRest();
+		rest.auth.requestToken(null, null, function(err, tokenDetails) {
+			if(err) {
+				test.ok(false, displayError(err));
+				closeAndFinish(test, realtime);
+				return;
+			}
+
+			var authPath = echoServer + "/?type=json&body=" + encodeURIComponent(JSON.stringify(tokenDetails));
+
+			realtime = helper.AblyRealtime({ authUrl: authPath });
+
+			realtime.connection.on('connected', function() {
+				test.ok(true, 'Connected to Ably using authUrl with TokenDetails JSON payload');
+
+				/* replace the authUrl and reauth */
+				realtime.auth.authorize(null, {authUrl: echoServer + '/respondwith?status=403'}, function(err, tokenDetails) {
+					test.equal(err && err.statusCode, 403, 'Check err statusCode');
+					test.equal(err && err.code, 40300, 'Check err code');
+					/* auth endpoints don't envelope, so this won't work with jsonp */
+					if(helper.bestTransport !== 'jsonp') {
```

Comment:
This comparison looks a little suspect to me but then I don't know the ins and outs of this test suite. It seems `bestTransport` is more used in tests to feed in to a next stage (common pattern being `realtime = helper.AblyRealtime({transports: [helper.bestTransport]})`, rather than for validation, so this just looks a bit odd. But, I don't know enough to be opinionated around whether it could or should be improved!